web/src/settings_config.ts

```diff
@@ -735,6 +738,8 @@ const other_email_settings = [
     "enable_marketing_emails",
 ];
 
+const topic_notifications = ["automatically_follow_topics_where_mentioned"];
+
```

Comment:
It'd be a bit confusing to keep only `automatically_follow_topics_where_mentioned` in `topic_mentions`. We have `automatically_follow_topics_policy` and `automatically_unmute_topics_in_muted_streams_policy`

We can achieve the expected behavior similar to `automatically_follow_topics_policy`.

But a broad question would be:
Do we plan to keep all of our settings following the `all_notifications` framework. I see `desktop_icon_count_display`, `realm_name_in_email_notifications_policy` and a few other settings not a part of that framework. Is this intended behavior?

@timabbott your thoughts?

Reply:
Yeah, this should sit next to `automatically_follow_topics_where_mentioned` in the system.

@sahil839 can probably better comment on how we want to organize these, but I think the intent was just to organize these variables to align with the sections in the UI.

Reply:
Oh, I see, this is working around the fact that the system only support checkboxes, not select settings;  if/when we improve the framework to do that, this will get cleaner.

Yeah this is probably fine.

Reply:
I think the setting can still be kept in `other_notification_settings` list which contains `automatically_follow_topics_policy` since that already has `notiifcation_sound` which is a select setting and can handle this as well.

Reply:
I have removed "automatically_follow_topics_where_mentioned" setting from ```topic_notifications```.